pyEpiabm/pyEpiabm/sweep/travel_sweep.py

```diff
@@ -0,0 +1,251 @@
+                    age_group=age)
+        else:
+            # Age not used in model
+            if isinstance(asymp_prop, list):
+                asymp_prop = asymp_prop[0]
+            number_indiv_InfectedAsympt = \
+                math.floor(asymp_prop * number_individuals_introduced)
+            number_indiv_InfectedMild = number_individuals_introduced - \
+                number_indiv_InfectedAsympt
+            self.initial_microcell.add_people(
+                    number_indiv_InfectedAsympt,
+                    status=InfectionStatus.InfectASympt)
+            self.initial_microcell.add_people(
+                    number_indiv_InfectedMild,
+                    status=InfectionStatus.InfectMild)
+
+        for person in self.initial_cell.persons:
+            # Assign introduced individuals a time to stay
+            person.travel_end_time = time + random.randint(
+                self.travel_params["duration_travel_stay"][0],
+                self.travel_params["duration_travel_stay"][1])
+            # Assigns them their next infection status and the time of
+            # their next status change. Also updates their infectiousness.
+            person.next_infection_status = InfectionStatus.Recovered
+            HostProgressionSweep.set_infectiousness(person, time)
+            HostProgressionSweep().update_time_status_change(person, time)
+
+    def assign_microcell_household(self, number_individuals_introduced):
+        """
+        Assign introduced individuals to microcells based on population
+        density of micorcells. Takes a number of microcells equal to the
+        the number_individuals_introduced (or max number of microcells)
+        and assigns individuals randomly to one of these microcells,
+        such that individuals can end up in the same microcell.
+
+        Parameters
+        ----------
+        number_individuals_introduced: int
+            Infected individuals added to population at certain time step
+
+        """
+        num_microcells_population = sum(len(cell.microcells) for cell in
+                                        self._population.cells)
+        microcells_to_choose_dict = {}
+        for i in range(min(number_individuals_introduced,
+                           num_microcells_population)):
+            # initialise with default microcell
+            microcells_to_choose_dict[Microcell(self.initial_cell)] = 0
+        # Find highest density microcells
+        for possible_cell in self._population.cells:
+            for possible_microcell in possible_cell.microcells:
+                density_microcells_list = list(
+                    microcells_to_choose_dict.values())
+                if min(density_microcells_list) < len(
+                        possible_microcell.persons):
+                    # remove old object
+                    microcells_to_choose_dict.pop(list(
+                        microcells_to_choose_dict.keys())[
```

Comment:
I do need explicit list conversion, otherwise I get the error: `TypeError: 'dict_keys' object is not subscriptable` and `AttributeError: 'dict_values' object has no attribute 'index'`

Reply:
My apologies - I thought these returned a list but it is actually a view object! Keep the list conversion!